File: TheWorld/wwwroot/js/tripsController.js
// tripController.js
(function () {
  "use-strict";

  //Getting the existing module
  angular.module("app-trips")
    .controller("tripsController", tripsController);

  function tripsController($http) {
    var vm = this;
    //vm.name = "khanhdd";
    vm.trips = [];

    var errorMessage = "";

    vm.newTrip = {};
    vm.isBusy = false;

    $http.get("/api/trips")
      .then(function (response) {
        angular.copy(response.data, vm.trips);
      },
      function (error) {
        vm.errorMessage = "Failed to load data: " + error;
      }).finally(function () {
        //vm.isBusy = false;
      })

    vm.addTrip = function () {
      vm.isBusy = true;
      $http.post("/api/trips", vm.newTrip)
        .then(function (response) {
          vm.trips.push(response.data);
          vm.newTrip = {};
        },
        function (error) {
          vm.errorMessage = "Failed to save data: " + error;
        })
        .finally(function () { vm.isBusy = false; });
    }


  }
})();

File: TheWorld/Api/TripsController.cs
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using TheWorld.Models;
using TheWorld.ViewModels;

namespace TheWorld.Api
{
    [Route("api/trips")]
    [Authorize]
    public class TripsController : Controller
    {
        private IWorldRepository _repository;
        private ILogger _logger;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }

        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {
                //if (false) BadRequest("Bad thing happened");
                //else
                //var result = _repository.GetAllTrips();
                var result = _repository.GetTripsByUserName(User.Identity.Name);
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(result));
            }
            catch (Exception ex)
            {
                //Logging
                _logger.LogError($"Failed to get All trips: {ex}");
                return BadRequest("Error occurred");
            }
        }

        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody]TripViewModel theTrip)
        {
            //if (false) BadRequest("Bad thing happened");
            //else
            if (ModelState.IsValid)
            {
                var newTrip = Mapper.Map<Trip>(theTrip);
                newTrip.UserName = User.Identity.Name;
                _repository.AddTrip(newTrip);
                if (await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
                }
            }
            return BadRequest("Failed to save the trip");
        }
    }
}
